File: Practica-3/Cola.java
package cola;

class Cola {
	private long[] arreglo;
	private int inicio;
	private int fin;
	private int n;
	
	public Cola(int n) {
		this.arreglo = new long[n];
		this.n = n;
		this.inicio = 0;
		this.fin= 0;
	}
	public void insert(long e) {
		if (isFull()) {
			System.out.println("Cola llena");
		} else {
			this.fin = this.fin + 1;
			this.arreglo[this.fin] = e;
		}
	}
	public long remove() {
		if (isEmpty()){
			System.out.println("cola vacia");
			return -1;
		} else {
			this.inicio = this.inicio + 1;
			long dato = this.arreglo[this.inicio];
			if(this.inicio == this.fin) {
				this.inicio = 0;
				this.fin = 0;
			}
			return dato;
		}
	}
	public long peek() {
		return this.arreglo[this.inicio + 1];
		
	}
	public boolean isEmpty() {
		return (this.inicio == 0 && this.fin == 0);
	}
	public boolean isFull() {
		return (this.fin == this.n);
	}
	public int size() {
		return (this.fin - this.inicio);
	}
	
	public static void main(String[] args) {
		Cola z = new Cola(9);
		z.insert(9);
		z.insert(4);
		z.insert(2);
		z.insert(1);
		z.insert(21);
		z.insert(0);
		z.insert(0);
		z.insert(0);
		
			
		System.out.println(z.size());
		System.out.println("------------------------------");
		
		while (! z.isEmpty()) {
			System.out.println(z.remove());
			
		}
		
		//z.mostrar(z);
		}
	}

File: Practica-3/Cola.py
class Cola:
    def __init__(self, n):
        self.__arreglo = [0] *(n +1)
        self.__n = n
        self.__inicio = 0
        self.__fin = 0
    def insert(self, e):
        if (self.__fin == self.__n):
            print("cola llena!!")
            return None
        else:
            self.__fin = self.__fin + 1
            self.__arreglo[self.__fin] = e

    def remove(self):
        if self.isEmpty():
            print("cola vacia")
            return None
        else:
            self.__inicio = (self.__inicio + 1)
            valor = self.__arreglo[self.__inicio]
            
            if (self.__inicio == self.__fin):
                self.__inicio = 0
                self.__fin = 0
            return valor
    def peek(self):
        return self.__arreglo[self.__inicio + 1]
    #--------------
    def isEmpty(self):
        return self.__fin == 0 and self.__inicio == 0
    def isFull(self):
        return self.__n == self.__fin
    def size(self):
        aux = Cola(self.__n)
        cont = 0
        while(not self.isEmpty()):
            dato = self.remove()
            aux.insert(dato)
            cont = cont + 1
        while(not aux.isEmpty()):
            dato = aux.remove()
            self.insert(dato)
        return cont
    def nroElem2(self): # size
        return self.__fin - self.__inicio
    
    


c = Cola(20)
c.insert(6)
c.insert(8)
c.insert(98)
c.insert(3)
c.insert(0)
c.insert(7)
c.remove()

print(c.size())

aux = Cola(20)

while (not c.isEmpty()):
    dat = c.remove()
    aux.insert(dat)
    print(dat, end=" ")
while(not aux.isEmpty()):
    dat = aux.remove()
    c.insert(dat)
        